pkg/api-server/inspect_endpoints.go

```diff
@@ -66,24 +69,31 @@ func addInspectEndpoints(
 
 	if cfg.Mode != config_core.Global {
 		ws.Route(
-			ws.GET("/meshes/{mesh}/dataplanes/{dataplane}/xds").To(inspectDataplaneXDS(envoyAdminClient, rm, cfg.GetEnvoyAdminPort())).
+			ws.GET("/meshes/{mesh}/dataplanes/{dataplane}/xds").To(inspectDataplaneXDS(envoyAdminClient, configDumpAccess, rm, cfg.GetEnvoyAdminPort())).
 				Doc("inspect dataplane XDS configuration").
 				Param(ws.PathParameter("mesh", "mesh name").DataType("string")).
 				Param(ws.PathParameter("dataplane", "dataplane name").DataType("string")),
 		)
 
 		ws.Route(
-			ws.GET("/zoneingresses/{zoneingress}/xds").To(inspectZoneIngressXDS(envoyAdminClient, rm, cfg.Multizone.Zone.Name, cfg.GetEnvoyAdminPort())).
+			ws.GET("/zoneingresses/{zoneingress}/xds").To(inspectZoneIngressXDS(envoyAdminClient, configDumpAccess, rm, cfg.Multizone.Zone.Name, cfg.GetEnvoyAdminPort())).
 				Doc("inspect zone ingresses XDS configuration").
 				Param(ws.PathParameter("zoneingress", "zoneingress name").DataType("string")),
 		)
 	} else {
+		methodNotAllowed := func(_ *restful.Request, response *restful.Response) {
```

Comment:
nit: why lambda? can this be a regular function?

Reply:
it relates to https://github.com/kumahq/kuma/pull/3768#discussion_r795817086, we have a huge package `api_server` with a lot of stuff and the name of the method was pretty generic, so it's lambda just to cut the scope of the function